### PR TITLE
zsys/logging: use zsock API for logging, zsock_attach syntax for logging endpoints

### PR DESCRIPTION
**Pull request by mhaberler**
see also: http://lists.zeromq.org/pipermail/zeromq-dev/2015-May/028953.html

### CODE REVIEW

**Comment by hintjens** [comment]
Travis reports:

    lt-czmq_selftest: src/zsys.c:1400: zsys_set_logsender: Assertion `rc == 0' failed.

Can you check that?

**Comment by mhaberler** [comment]
yes, will do

**Comment by mhaberler** [comment]
so far I'm unable to reproduce the Travis error, moving to different machine now

**Comment by mhaberler** [comment]
interesting, on debian jessie I do have libtool installed, but autogen.sh fails:

mah@next:~/czmq$ apt-show-versions |grep libtool
libtool:all/jessie 2.4.2-1.11 uptodate

since /usr/bin/libtool is not part of the libtool package anymore (?): https://packages.debian.org/jessie/all/libtool/filelist

**Comment by Asmod4n** [comment]
try libtool-bin

**Comment by mhaberler** [comment]
ah, thanks, that did the trick

**Comment by mhaberler** [comment]
hm,  on jessie the zdir selftest loops, using the sock libzmq package:

libzmq3:i386/jessie 4.0.5+dfsg-2+deb8u1 uptodate
libzmq3-dev:i386/jessie 4.0.5+dfsg-2+deb8u1 uptodate

very weird, make check doesnt even get to the failure Pieter reported

where can I find details how the travis check is set up?

**Comment by mhaberler** [comment]
the loop in the zdir selftest is caused by a symlink pointing to the same directory:

ln -s xenomai .

peng!

**Comment by hintjens** [comment]
What I do in such cases is debug the error in Travis... I.e. add printfs
into your code, push it, enable the build on your repo, and get the output,
repeat until you have figured out the problem.

On Thu, May 21, 2015 at 12:30 PM, Michael Haberler <notifications@github.com

**Comment by hintjens** [comment]
hmm

On Thu, May 21, 2015 at 12:33 PM, Michael Haberler <notifications@github.com

**Comment by mhaberler** [comment]
I disabled the zdir test, zsys passes on Jessie. duh.

**Comment by mhaberler** [comment]
very weird, by using the ci_build.sh script I do not get an assert but a hang:

 * zsys: 
I: 15-05-21 12:53:16 host name is nwheezy.mah.priv.at
I: 15-05-21 12:53:16 system limit is 1024 ZeroMQ sockets
E: (czmq_selftest) 15-05-21 12:53:16 This is an error message
W: (czmq_selftest) 15-05-21 12:53:16 This is a warning message
N: (czmq_selftest) 15-05-21 12:53:16 This is a notice message
I: (czmq_selftest) 15-05-21 12:53:16 This is a info message
D: (czmq_selftest) 15-05-21 12:53:16 This is a debug message
I: (hello, world) 15-05-21 12:53:16 This is a info message
D: (hello, world) 15-05-21 12:53:16 This is a debug message


root@nwheezy:/home/mah/src/czmq# gdb -p 757
GNU gdb (GDB) 7.4.1-debian
Copyright (C) 2012 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "i486-linux-gnu".
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Attaching to process 757
Reading symbols from /home/mah/src/czmq/src/.libs/lt-czmq_selftest...done.
Reading symbols from /home/mah/src/czmq/src/.libs/libczmq.so.3...done.
Loaded symbols for /home/mah/src/czmq/src/.libs/libczmq.so.3
Reading symbols from /usr/lib/i386-linux-gnu/libzmq.so.3...(no debugging symbols found)...done.
Loaded symbols for /usr/lib/i386-linux-gnu/libzmq.so.3
Reading symbols from /lib/i386-linux-gnu/i686/cmov/libc.so.6...Reading symbols from /usr/lib/debug/lib/i386-linux-gnu/i686/cmov/libc-2.13.so...done.
done.
Loaded symbols for /lib/i386-linux-gnu/i686/cmov/libc.so.6
Reading symbols from /usr/lib/libsodium.so.10...(no debugging symbols found)...done.
Loaded symbols for /usr/lib/libsodium.so.10
Reading symbols from /lib/i386-linux-gnu/i686/cmov/librt.so.1...Reading symbols from /usr/lib/debug/lib/i386-linux-gnu/i686/cmov/librt-2.13.so...done.
done.
Loaded symbols for /lib/i386-linux-gnu/i686/cmov/librt.so.1
Reading symbols from /lib/i386-linux-gnu/i686/cmov/libpthread.so.0...Reading symbols from /usr/lib/debug/lib/i386-linux-gnu/i686/cmov/libpthread-2.13.so...done.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/i386-linux-gnu/i686/cmov/libthread_db.so.1".
[New Thread 0x408b9b70 (LWP 815)]
[New Thread 0x406b8b70 (LWP 814)]
done.
Loaded symbols for /lib/i386-linux-gnu/i686/cmov/libpthread.so.0
Reading symbols from /usr/lib/i386-linux-gnu/libstdc++.so.6...(no debugging symbols found)...done.
Loaded symbols for /usr/lib/i386-linux-gnu/libstdc++.so.6
Reading symbols from /lib/i386-linux-gnu/i686/cmov/libm.so.6...Reading symbols from /usr/lib/debug/lib/i386-linux-gnu/i686/cmov/libm-2.13.so...done.
done.
Loaded symbols for /lib/i386-linux-gnu/i686/cmov/libm.so.6
Reading symbols from /lib/i386-linux-gnu/libgcc_s.so.1...(no debugging symbols found)...done.
Loaded symbols for /lib/i386-linux-gnu/libgcc_s.so.1
Reading symbols from /lib/ld-linux.so.2...(no debugging symbols found)...done.
Loaded symbols for /lib/ld-linux.so.2
Reading symbols from /lib/i386-linux-gnu/i686/cmov/libnss_files.so.2...Reading symbols from /usr/lib/debug/lib/i386-linux-gnu/i686/cmov/libnss_files-2.13.so...done.
done.
Loaded symbols for /lib/i386-linux-gnu/i686/cmov/libnss_files.so.2
0x4001e424 in __kernel_vsyscall ()
(gdb) backtrace 
#0  0x4001e424 in __kernel_vsyscall ()
#1  0x40258286 in *__GI___poll (fds=0x402eeff4, nfds=1, timeout=-1)
    at ../sysdeps/unix/sysv/linux/poll.c:87
#2  0x40116094 in ?? () from /usr/lib/i386-linux-gnu/libzmq.so.3
#3  0x400ff13e in ?? () from /usr/lib/i386-linux-gnu/libzmq.so.3
#4  0x4011729d in ?? () from /usr/lib/i386-linux-gnu/libzmq.so.3
#5  0x40117a97 in ?? () from /usr/lib/i386-linux-gnu/libzmq.so.3
#6  0x4012dcfc in ?? () from /usr/lib/i386-linux-gnu/libzmq.so.3
#7  0x4012ddcb in zmq_recvmsg () from /usr/lib/i386-linux-gnu/libzmq.so.3
#8  0x4005eed4 in zstr_recv (source=source@entry=0x40b025a0) at src/zstr.c:67
#9  0x40061f8f in zsys_test (verbose=verbose@entry=true) at src/zsys.c:1681
#10 0x08048f82 in main (argc=2, argv=0xbfc65b14) at src/czmq_selftest.c:63
(gdb)

**Comment by mhaberler** [comment]
oh nice, changing serverish to false on zsock_attach() did the trick!

**Comment by mhaberler** [comment]
Any reason not to merge this patch?

File: src/zsys.c [hunk 1]
```diff
@@ -1392,17 +1392,16 @@ zsys_set_logsender (const char *endpoint)
     if (endpoint) {
         //  Create log sender if needed
         if (!s_logsender) {
-            s_logsender = zsys_socket (ZMQ_PUB, NULL, 0);
+            s_logsender = zsock_new_pub(NULL);
```
Comment (Asmod4n) [inline src/zsys.c]: line number 1399 is not needed, you can supply the endpoint here.

**Comment by hintjens** [comment]
I was just waiting for you to confirm that it all worked... thanks.

File: src/zsys.c (hunk 1 above)
Comment (mhaberler) [inline src/zsys.c]: I do not understand your reference to "here".

If you mean zsock_attach() is not needed - it is, if you want to use the ">"/"@" connect/bind syntax for the endpoint which zsys_socket() does not support.

**Comment by mhaberler** [comment]
sorry for being unclear. Yes - it works just grandiose as planned ;)

**Pull request merged**